### PR TITLE
Fix/error message

### PR DESCRIPTION
**Pull request by tshao87**
Please make sure to follow the [DEV guidelines](https://gen3.org/resources/developer/dev-introduction/) before asking for review.

### Improvements
- improved message output
- rename `upload-old` to `upload-single` 
- removed redundant codes

### CODE REVIEW

File: gen3-client/g3cmd/upload-manifest.go [hunk 1]
```diff
@@ -114,12 +114,13 @@ func init() {
...
-		Short:      "upload files from a specified manifest",
-		Long:       `Gets a presigned URL for a file from a GUID and then uploads the specified file.`,
-		Example:    `./gen3-client upload-manifest --profile=<profile-name> --manifest=<path-to-manifest/manifest.json> --upload-path=<path-to-file-dir/>`,
-		Deprecated: `use "./gen3-client upload" instead.`,
+		Use:     "upload-manifest",
+		Short:   "upload files from a specified manifest",
+		Long:    `Gets a presigned URL for a file from a GUID and then uploads the specified file.`,
+		Example: `./gen3-client upload-manifest --profile=<profile-name> --manifest=<path-to-manifest/manifest.json> --upload-path=<path-to-file-dir/>`,
 		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println("Notice: this is the old upload flow which requires the user to provide GUIDs. If your common supports the new upload flow, consider to use \"./gen3-client upload\" instead.")
```
Comment (abgeorge7) [inline gen3-client/g3cmd/upload-manifest.go]: Nice! I'm wondering if there are more descriptive terms for "old" and "new" upload flow, not sure if the user will understand what those are referring to?

File: gen3-client/jwt/functions.go [hunk 2]
```diff
@@ -99,24 +98,24 @@ func (f *Functions) ParseFenceURLResponse(resp *http.Response) (JsonMessage, err
 	msg := JsonMessage{}
 
 	if resp == nil {
-		return msg, nil
+		return msg, errors.New("Nil response received")
 	}
 
 	if resp.StatusCode == 401 {
-		return msg, errors.New("401 Unauthorized error has occured! Something went wrong during authentication, please check your configuration and/or credentials.")
+		return msg, errors.New("401 Unauthorized error has occured! Something went wrong during authentication, please check your configuration and/or credentials")
```
Comment (abgeorge7) [inline gen3-client/jwt/functions.go]: picky but should be "occurred"

File: gen3-client/jwt/functions.go [hunk 3]
```diff
@@ -126,26 +125,26 @@ func (f *Functions) ParseFenceURLResponse(resp *http.Response) (JsonMessage, err
...
+func (f *Functions) DoRequestWithSignedHeader(profile string, configFileType string, endpointPostPrefix string, contentType string, bodyBytes []byte) (string, string, error) {
 	/*
 		Do request with signed header. User may have more than one profile and use a profile to make a request
 	*/
 	var resp *http.Response
 
 	cred := f.Config.ParseConfig(profile)
 	if cred.APIKey == "" && cred.AccessKey == "" && cred.APIEndpoint == "" {
-		return "", "", errors.New("No credential found!")
+		return "", "", errors.New("No credential found")
```
Comment (abgeorge7) [inline gen3-client/jwt/functions.go]: Maybe this one could be more descriptive as well?

File: gen3-client/g3cmd/upload-manifest.go (hunk 1 above)
Comment (tshao87) [reply gen3-client/g3cmd/upload-manifest.go]: Right, I'm also wondering the same thing. For now even ourselves are using terms "old upload flow" and "new upload flow" and we should have some better names to differentiate them 🤔.

What do you think if we call the command to run the new upload flow a "simplified upload command"? Although it is not really that simple 🤔

File: gen3-client/jwt/functions.go (hunk 2 above)
Comment (tshao87) [reply gen3-client/jwt/functions.go]: fixed! good catch

File: gen3-client/g3cmd/upload-manifest.go (hunk 1 above)
Comment (abgeorge7) [reply gen3-client/g3cmd/upload-manifest.go]: @zflamig is there a reason we should even have two flows at all? can we just get rid of the old one?

Comment (abgeorge7) [reply gen3-client/g3cmd/upload-manifest.go]: I think we can go with those names then 👍

**Pull request merged**